mars/core/entity/tileables.py

```diff
@@ -81,6 +87,7 @@ def tile(
                 # without iterative tiling
                 tiled_result = tile_handler(op)
         finally:
+            cls._check_shuffle(tiled_result)
```

Comment:
Many chunks are not contained in tileable.chunks, such as `df.describe`.  We can't get all generated chunks here.
![image](https://user-images.githubusercontent.com/12445254/174769022-f6ca2b7f-3c10-4238-b3c0-76d2e67485a7.png)

Reply:
Just check the `mars/mars/services/task/supervisor/tests/task_preprocessor.py` to see if it's possible to integrate the check into it, because the check logic is sealed in the tests module instead of putting it directly into the normal code which looks really ugly.

Reply:
Actually I tried to add the checks to `CheckedTaskPreprocessor` at first, but the `CheckedTaskPreprocessor` only have `ChunkGraph` as inputs. Since some reducers are not contained in the chunk graph. Add checks here won't cover all cases. I agree with you add the checks in normal code is not elegant. But seems we can't find a place to get all tiled reducers. The checks here just to capture operands implementation error.

Reply:
The checks here just to capture operands implementation error. The normal codepath doesn't need this.

Reply:
Moved the checks to `CheckedTaskPreprocessor`